Backend/app/rider-platform/rider-app/Main/src/Storage/Tabular/Booking.hs

```diff
@@ -95,53 +94,53 @@ instance FromTType FullBookingT Domain.Booking where
       RentalDetailsT rentalSlabT -> Domain.RentalDetails <$> fromTType rentalSlabT
       DriverOfferDetailsT toLocT -> Domain.DriverOfferDetails <$> buildOneWayDetails toLocT
       OneWaySpecialZoneDetailsT toLocT -> Domain.OneWaySpecialZoneDetails <$> buildOneWaySpecialZoneDetails toLocT
-    return $
-      Domain.Booking
+    pure
+      DB.Booking
         { id = Id id,
           bppBookingId = Id <$> bppBookingId,
           paymentMethodId = fromKey <$> paymentMethodId,
           riderId = fromKey riderId,
           quoteId = fromKey <$> quoteId,
-          providerUrl = pUrl,
           merchantId = fromKey merchantId,
           estimatedFare = roundToIntegral estimatedFare,
           discount = roundToIntegral <$> discount,
           estimatedTotalFare = roundToIntegral estimatedTotalFare,
+          providerUrl = pUrl,
           ..
         }
     where
       buildOneWayDetails toLocT = do
-        toLocation <- fromTType toLocT
+        toLocation <- mapM fromTType toLocT
         distance' <- distance & fromMaybeM (InternalError "distance is null for one way booking")
         pure
           Domain.OneWayBookingDetails
             { toLocation,
               distance = distance'
             }
       buildOneWaySpecialZoneDetails toLocT = do
-        toLocation <- fromTType toLocT
+        toLocation <- mapM fromTType toLocT
         distance' <- distance & fromMaybeM (InternalError "distance is null for one way booking")
         pure
           Domain.OneWaySpecialZoneBookingDetails
             { distance = distance',
               ..
             }
 
-instance ToTType FullBookingT Domain.Booking where
-  toTType Domain.Booking {..} = do
-    let (fareProductType, bookingDetailsT, toLocationId, distance, rentalSlabId, otpCode) = case bookingDetails of
+instance ToTType FullBookingT DB.Booking where
+  toTType DB.Booking {..} = do
+    let (fareProductType, bookingDetailsT, _, distance, rentalSlabId, otpCode) = case bookingDetails of
           Domain.OneWayDetails details -> do
-            let toLocT = toTType details.toLocation
-            (DQuote.ONE_WAY, OneWayDetailsT toLocT, Just . toKey $ details.toLocation.id, Just details.distance, Nothing, Nothing)
+            let toLocT = map toTType details.toLocation
+            (DQuote.ONE_WAY, OneWayDetailsT toLocT, [], Just details.distance, Nothing, Nothing)
```

Comment:
why do you return empty list everywhere?

Reply:
because it is created separately while creating booking,ride and locations